src/json-rpc-handlers/dwn/process-message.ts

```diff
@@ -0,0 +1,104 @@
+import type { Readable } from 'readable-stream';
+import type { JsonRpcHandler, HandlerResponse } from '../../lib/json-rpc-router.js';
+
+import busboy from 'busboy';
+import EventEmitter from 'events';
+
+import { Dwn } from '@tbd54566975/dwn-sdk-js';
+import { base64url } from 'multiformats/bases/base64';
+import { DataStream } from '@tbd54566975/dwn-sdk-js';
+import { v4 as uuidv4 } from 'uuid';
+
+import { JsonRpcErrorCodes, createJsonRpcErrorResponse, createJsonRpcSuccessResponse } from '../../lib/json-rpc.js';
+
+const emitter = new EventEmitter();
+
+export const handleDwnProcessMessage: JsonRpcHandler = async (dwnRequest, context) => {
+  const { dwn } = context;
+  const requestId = dwnRequest.id ?? uuidv4();
+  const { target, message } = dwnRequest.params;
+
+  if (context.contentType === 'application/octet-stream') {
+    return await _processDwnMessage(dwn, requestId, target, message, context.request as any);
+  } else if (context.contentType === 'multipart/form-data') {
+    // there's no choice other than to return a promise here because we have to wait until the multipart
+    // request body is fully consumed.
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(resolve => {
+      const bb = busboy({
+        headers : context.request.headers,
+        limits  : {
+          fields : 0,
+          files  : 1
+        }
+      });
+
+      let { params } = dwnRequest;
+
+      emitter.once(requestId, responsePayload => {
```

Comment:
do we actually really need to use an `EventEmitter`?  can we just call `resolve` directly?
```
bb.on('file', async (_name, stream, _info) => {
  const responsePayload = await _processDwnMessage(dwn, requestId, params.target, params.message, <any>stream);
  resolve(responsePayload);
});
```
im pretty sure that `Promise` will just silently noop if `resolve` is called more than once